src/frequenz/sdk/data_ingestion/resampling/component_metrics_resampling_actor.py

```diff
@@ -0,0 +1,175 @@
+        self._output_senders: Dict[str, Sender[Sample]] = {}
+        self._resampling_timer = Timer(interval=self._resampling_period_s)
+
+    async def subscribe(self, request: ComponentMetricRequest) -> None:
+        """Subscribe for data for a specific time series.
+
+        Args:
+            request: subscription request for a specific component metric
+        """
+        channel_name = request.get_channel_name()
+
+        data_source_request = dataclasses.replace(request, **dict(namespace="Source"))
+        data_source_channel_name = data_source_request.get_channel_name()
+        if channel_name not in self._input_receivers:
+            await self._subscription_sender.send(data_source_request)
+            receiver: Receiver[Sample] = self._channel_registry.get_receiver(
+                data_source_channel_name
+            )
+            self._input_receivers[data_source_channel_name] = receiver
+            self._resampler.add_time_series(time_series_id=data_source_channel_name)
+
+        if channel_name not in self._output_senders:
+            sender: Sender[Sample] = self._channel_registry.get_sender(channel_name)
+            # This means that the `sender` will be sending samples to the channel with
+            # name `channel_name` based on samples collected from the channel named
+            # `data_source_channel_name`
+            self._output_senders[data_source_channel_name] = sender
+
+    def is_sample_valid(self, sample: Sample) -> bool:
+        """Check if the provided sample is valid.
+
+        Args:
+            sample: sample to be validated
+
+        Returns:
+            True if the sample is valid, False otherwise
+        """
+        if sample.value is None or math.isnan(sample.value):
+            return False
+        return True
+
+    async def run(self) -> None:
+        """Run the actor.
+
+        Raises:
+            ConnectionError: When the provider of the subscription channel closes the
+                connection
+        """
+        while True:
+            select = Select(
+                resampling_timer=self._resampling_timer,
+                subscription_receiver=self._subscription_receiver,
+                component_data_receiver=MergeNamed(**self._input_receivers),
+            )
+            while await select.ready():
+                if msg := select.subscription_receiver:
+                    if msg.inner is None:
+                        raise ConnectionError(
+                            "Subscription channel connection has been closed!"
+                        )
+                    await self.subscribe(request=msg.inner)
+                    # Breaking out from the loop is required to regenerate
+                    # component_data_receivers to be able to fulfil this
+                    # subscription (later can be optimized by checking if
+                    # an output channel already existed in the `subscribe()` method)
+                    break
```

Comment:
This handler should be at the end of the loop to avoid losing messages if there were other channels with ready messages.

Reply:
Done